PurchasesCoreSwift/Misc/SystemInfo.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  SystemInfo.swift
+//  PurchasesCoreSwift
+//
+//  Created by Joshua Liebowitz on 6/29/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+import UIKit
+#elseif os(watchOS)
+import UIKit
+import WatchKit
+#elseif os(macOS)
+import AppKit
+#endif
+
+public typealias RCSystemInfo = SystemInfo
+
+@objc(RCSystemInfo) open class SystemInfo: NSObject {
+
+    #if targetEnvironment(macCatalyst)
+    static let PLATFORM_HEADER = "uikitformac"
+    #elseif os(iOS)
+    static let PLATFORM_HEADER = "iOS"
+    #elseif os(watchOS)
+    static let PLATFORM_HEADER = "watchOS"
+    #elseif os(tvOS)
+    static let PLATFORM_HEADER = "tvOS"
+    #elseif os(macOS)
+    static let PLATFORM_HEADER = "macOS"
+    #endif
+
+    public enum SystemInfoError: Error {
+        case invalidInitializationData
+    }
+
+    private static let defaultServerHostName = "https://api.revenuecat.com"
+
+    @objc public var finishTransactions: Bool
+    @objc public let platformFlavor: String
+    @objc public let platformFlavorVersion: String?
+    @objc public static var forceUniversalAppStore: Bool = false
+    @objc public static var isSandbox: Bool {
+        let url = Bundle.main.appStoreReceiptURL
+        guard let url = url else {
+            return false
+        }
+
+        let receiptURLString = url.path
+        return receiptURLString.contains("sandboxReceipt")
+    }
+    @objc public static func frameworkVersion() -> String { // TODO: automate the setting of this, if it hasn't been.
+        return "3.12.0-SNAPSHOT"
+    }
+
+    @objc public static func systemVersion() -> String {
+        return ProcessInfo().operatingSystemVersionString
+    }
+
+    @objc public static func appVersion() -> String {
+        return Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
+    }
+
+    @objc public static func buildVersion() -> String {
+        return Bundle.main.infoDictionary?["CFBundleVersion"] as? String ?? ""
+    }
+
+    @objc public static func platformHeader() -> String {
```

Comment:
these might make more sense as computed vars now or even lets, if they're available as soon as the app starts

Reply:
Yeah, that was my first pass, turns out this is public API and it uses `()`. We could make vars and deprecate these funcs, though.

Reply:
Since it's not public, I'll switch them back to vars.